debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/J9ConstantHelper.java

```diff
@@ -26,6 +26,25 @@
 
 public final class J9ConstantHelper {
 
+	private static Field getStaticFinalField(Class<?> clazz, String name) throws NoSuchFieldException, SecurityException {
+		Field field = clazz.getField(name);
+		int modifiers = field.getModifiers();
+
+		// check that the field has the expected modifiers and type
+		if (!Modifier.isPublic(modifiers) || !Modifier.isStatic(modifiers) || !Modifier.isFinal(modifiers)) {
+			String message = String.format("%s.%s is not public static final", clazz.getName(), name);
+
+			throw new IllegalArgumentException(message);
+		}
+
+		if (field.getType() != long.class) {
```

Comment:
This helper method needs a parameter to specify the expected field type: `long` is wrong for the use in `getBoolean` below.